client/src/pages/Admin.js

```diff
@@ -51,43 +60,94 @@ function AdminPage() {
 			}
 		}
 
+async function getAvailableModules(){
+  try {
+    const availableModule = await fetch("/api/listOfmodules");
+    const modulesData = await availableModule.json();
+    return modulesData;
+  } catch (error) {
+    console.error(error);
+    return "No Module is not avaialbe";
+  }
+}
+
+
+useEffect(() => {
+	getAvailableModules().then((modulesResult) => setModules(modulesResult));
```

Comment:
Why don't you put this in the other useEffect that you have?

Reply:
Delete one useEffect and only one useEffect is in use now.